src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt

```diff
@@ -256,6 +258,49 @@ class FunSpecTest {
       |""".trimMargin())
   }
 
+  @Test fun thisConstructorDelegate() {
+    val funSpec = FunSpec.constructorBuilder()
+        .addParameter("list", ParameterizedTypeName.get(List::class, Int::class))
+        .delegateConstructor(THIS, "list[0]", "list[1]")
+        .build()
+
+    assertThat(funSpec.toString()).isEqualTo("""
+      |constructor(list: kotlin.collections.List<kotlin.Int>) : this(list[0], list[1])
+      |""".trimMargin())
+  }
+
+  @Test fun superConstructorDelegate() {
+    val funSpec = FunSpec.constructorBuilder()
+        .addParameter("list", ParameterizedTypeName.get(List::class, Int::class))
+        .delegateConstructor(SUPER, "list[0]", "list[1]")
```

Comment:
would the API be simpler if there were just two methods, `thisConstructor` and `superConstructor` ?

Reply:
Then there’s no `Constructor` type, which seems like a simplification to me

Reply:
Or should these args go on `constructorBuilder()` above?

Reply:
Would it be clear enough if we drop the `delegate` part (i.e. `delegateThisConstructor()` vs `thisConstructor()`)? I agree with having two methods vs the enum param, that was what the initial version looked like (albeit a lot more verbose), and I guess I like it better. 

For the `constructorBuilder()` args, we'll need to have different versions of it for `this` and `super`, and I can't really think of good names. `thisConstructorBuilder()` and `superConstructorBuilder()`? And keeping the original `constructorBuilder()` with no args?